File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = [0, 0]
    word_dic = {}
    person_dic = dict.fromkeys(range(1, n+1), 0)
    person_idx = 2
    
    # 첫 번째 사람이 말한 거
    compare_word = words[0]
    word_dic[words[0]] = 1
    person_dic[1] = 1
    
    # 그 다음부터
    for word in words[1:]:
        person_dic[person_idx] += 1
        if compare_word[-1] != word[0] or word_dic.get(word) or len(word) == 1:
            answer = [person_idx, person_dic[person_idx]]
            break
        person_idx += 1
        if person_idx == (n+1):
            person_idx = 1
        compare_word = word
        word_dic[word] = 1

    return answer

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = {};
        int[] count = new int[n];
        Map<String, Boolean> exist = new HashMap<>();

        for(int i=0 ; i<words.length ; i++){
            String word = words[i];
            count[i % n]++;
            if((i > 0 && words[i - 1].charAt(words[i - 1].length() - 1) != word.charAt(0))
                || exist.containsKey(word)){
                return new int[]{(i % n) + 1, count[i % n]};
            }
            
            exist.put(word, true);
        }

        return new int[]{0, 0};
    }
}